### DISCUSSION TITLE
What about the "sandbox" alternative?

### DISCUSSION THREAD

**Discussion by ikas-mc** [microsoft/WindowsAppSDK#1071] [Q&A]
UWP applications work in  "sandbox"
But app use  WindowsAppSDK  has full permissions by default
What about the "sandbox" alternative?

**Comment by ghost** [comment]
this is a very good question for the reason that _it has a concern for security._ 
> UWP applications work in "sandbox"

The term AppContainer (MS engineer explaining [what it actually is and how it works](https://github.com/microsoft/WindowsAppSDK/discussions/410#discussioncomment-364688))  people often use exactly equals to the native applications sandbox in windows.

> But app use WindowsAppSDK has full permissions by default

assuming that you mean Full permissions = High IL apps/semi god mode, Currently windowsAppSDK produce windows apps that can only run in Medium IL. there is [a windows bug where WinRT apis doesn't work](https://github.com/microsoft/WindowsAppSDK/discussions/671#discussioncomment-610768) in High IL apps — yet. 

> What about the "sandbox" alternative?
 
there is no alternative, _AppContainer is a core NT Kernel feature and app models agonistic_.   The sandboxed environment of UWP is like trying to drive a car with your hands tied behind your back. Current AppContainer has  very basic features allowances and _too much crippled to do anything useful._  what they can do is to allow more features  to  be accessible right within the appContainer by brokering more and more features. 

Imagine using a closed source third party application you don't trust yet you installed for a specific purpose and it spies on your every keyboard strokes in the background. that's the whole situation right now with medium IL apps.

PM of WindowsAppSDK has _signaled_ the [AppContainer Improvements for win32 apps](https://twitter.com/andrewclinick/status/1412191931585597471) , not a promise per se but you can take that as a _logical hint_ for what to come.

**Reply by ikas-mc** [reply]
I have read related issues and benefited a lot,thanks
Hope AppContainer for win32 apps will be officially supported 
The old uwp is dying, and the new WindowsAppSDK application cannot replace uwp in security. This is the problem that needs to be faced now... 
There is no clear plan. Terrible
